File: mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MifosBaseFragment.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.mifosxdroid.core;

import android.app.Activity;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.appcompat.widget.Toolbar;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import com.mifos.mifosxdroid.R;
import com.mifos.utils.Network;

/**
 * @author fomenkoo
 */
public class MifosBaseFragment extends Fragment {

    private BaseActivityCallback callback;
    private Activity activity;
    private InputMethodManager inputManager;
    private MifosProgressBarHandler mMifosProgressBarHandler;

    @SuppressWarnings("deprecation")
    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        this.activity = activity;
        try {
            callback = (BaseActivityCallback) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString() + " must implement " +
                    "BaseActivityCallback");
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        inputManager = (InputMethodManager) getActivity().getSystemService(Context
                .INPUT_METHOD_SERVICE);
        mMifosProgressBarHandler = new MifosProgressBarHandler(getActivity());
    }

    public void showAlertDialog(String title, String message) {
        new MaterialDialog.Builder().init(getActivity())
                .setTitle(title)
                .setMessage(message)
                .setCancelable(false)
                .setPositiveButton(getStringMessage(R.string.dialog_action_ok), new
                        DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        dialog.dismiss();
                    }
                })
                .createMaterialDialog()
                .show();
    }

    public Toolbar getToolbar() {
        return ((MifosBaseActivity) getActivity()).getToolbar();
    }

    protected void showMifosProgressDialog() {
        showMifosProgressDialog("Working...");
    }

    protected void showMifosProgressDialog(String message) {
        if (callback != null)
            callback.showProgress(message);
    }

    protected void hideMifosProgressDialog() {
        if (callback != null)
            callback.hideProgress();
    }

    protected void logout() {
        callback.logout();
    }

    protected void setToolbarTitle(String title) {
        callback.setToolbarTitle(title);
    }

    public void hideKeyboard(View view) {
        inputManager.hideSoftInputFromWindow(view.getWindowToken(), InputMethodManager
                .RESULT_UNCHANGED_SHOWN);
    }

    protected void showMifosProgressBar() {
        mMifosProgressBarHandler.show();
    }

    protected void hideMifosProgressBar() {
        mMifosProgressBarHandler.hide();
    }

    protected String getStringMessage(int message) {
        return getResources().getString(message);
    }

    protected Boolean isOnline() {
        return Network.isOnline(getActivity());
    }
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MifosBaseFragment.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.mifosxdroid.core

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import com.mifos.mifosxdroid.R
import com.mifos.utils.LanguageHelper
import com.mifos.utils.Network

/**
 * @author fomenkoo
 */
open class MifosBaseFragment() : Fragment() {
    private var callback: BaseActivityCallback? = null
    private var activity: Activity? = null
    private var inputManager: InputMethodManager? = null
    private var mMifosProgressBarHandler: MifosProgressBarHandler? = null
    override fun onAttach(context: Context) {
        super.onAttach(LanguageHelper.onAttach(context))
        val activity = if (context is Activity) context else null
        this.activity = activity
        callback = try {
            activity as BaseActivityCallback?
        } catch (e: ClassCastException) {
            throw ClassCastException(
                activity.toString() + " must implement " +
                        "BaseActivityCallback"
            )
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        inputManager =
            requireActivity().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        mMifosProgressBarHandler = MifosProgressBarHandler(requireActivity())
    }

    fun showAlertDialog(title: String?, message: String?) {
        MaterialDialog.Builder().init(getActivity())
            .setTitle(title)
            .setMessage(message)
            .setCancelable(false)
            .setPositiveButton(getStringMessage(R.string.dialog_action_ok)) { dialog, id -> dialog.dismiss() }
            .createMaterialDialog()
            .show()
    }

    val toolbar: Toolbar?
        get() = (getActivity() as MifosBaseActivity).toolbar

    protected fun showMifosProgressDialog(message: String = "Working...") {
        if (callback != null) callback?.showProgress(message)
    }

    protected fun hideMifosProgressDialog() {
        if (callback != null) callback?.hideProgress()
    }

    protected fun logout() {
        callback?.logout()
    }

    protected fun setToolbarTitle(title: String) {
        callback?.setToolbarTitle(title)
    }

    fun hideKeyboard(view: View) {
        inputManager?.hideSoftInputFromWindow(
            view.windowToken,
            InputMethodManager.RESULT_UNCHANGED_SHOWN
        )
    }

    protected fun showMifosProgressBar() {
        mMifosProgressBarHandler?.show()
    }

    protected fun hideMifosProgressBar() {
        mMifosProgressBarHandler?.hide()
    }

    protected fun getStringMessage(message: Int): String {
        return resources.getString(message)
    }

    protected val isOnline: Boolean
        get() = Network.isOnline(requireContext())
}